service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/CopyGcsBucketDataStep.java

```diff
@@ -0,0 +1,404 @@
+    final BucketInputs destinationInputs = getDestinationInputs(flightContext);
+    logger.info(
+        "Starting data copy from source bucket \n\t{}\nto destination\n\t{}",
+        sourceInputs,
+        destinationInputs);
+
+    final String transferJobName = createTransferJobName();
+    final String controlPlaneProjectId =
+        Optional.ofNullable(ServiceOptions.getDefaultProjectId())
+            .orElseThrow(
+                () ->
+                    new IllegalStateException(
+                        "Could not determine default GCP control plane project ID."));
+    logger.info(
+        "Creating transfer job named {} in project {}", transferJobName, controlPlaneProjectId);
+
+    try {
+      final Storagetransfer storageTransferService = createStorageTransferService();
+
+      // Get the service account in the control plane project used by the transfer service to
+      // perform the actual data transfer. It's named for and scoped to the project.
+      final String transferServiceSAEmail = storageTransferService
+          .googleServiceAccounts()
+          .get(controlPlaneProjectId)
+          .execute()
+          .getAccountEmail();
+      logger.debug("Storage Transfer Service SA: {}", transferServiceSAEmail);
+      // Set source bucket permissions
+      addBucketRoles(sourceInputs, transferServiceSAEmail);
+
+      // Set destination roles
+      addBucketRoles(destinationInputs, transferServiceSAEmail);
+
+      final TransferJob transferJobInput =
+          new TransferJob()
+              .setName(transferJobName)
+              .setDescription("Terra Workspace Manager Clone GCS Bucket")
+              .setProjectId(controlPlaneProjectId)
+              .setSchedule(createScheduleRunOnceNow())
+              .setTransferSpec(
+                  createTransferSpec(
+                      sourceInputs.getBucketName(), destinationInputs.getBucketName()))
+              .setStatus(ENABLED_STATUS);
+      // Create the TransferJob for the associated schedule and spec in the correct project.
+      final TransferJob transferJobOutput =
+          storageTransferService.transferJobs().create(transferJobInput).execute();
+      logger.debug("Created transfer job {}", transferJobOutput);
+      // Job is now submitted with its schedule. We need to poll the transfer operations API
+      // for completion of the first transfer operation. The trick is going to be setting up a
+      // polling interval that's appropriate for a wide range of bucket sizes. Everything from millisecond
+      // to hours. The transfer operation won't exist until it starts.
+      final String operationName =
+          getLatestOperationName(storageTransferService, transferJobName, controlPlaneProjectId);
+
+      final StepResult operationResult =
+          getTransferOperationResult(storageTransferService, transferJobName, operationName);
+      removeBucketRoles(sourceInputs, transferServiceSAEmail);
```

Comment:
Should these be in the undo?

Hmm, I think that would require making a restructuring of steps to:
- get the SA, set the permissions with undo to unset them
- do the data copy
- unset them after successful transfer

Reply:
done